### PR TITLE
[1288] Updated course withdrawal

### PR DESCRIPTION
**Pull request by defong**
### Context

Course Withdrawal

### Changes proposed in this pull request
Removed the course code check method
Added additional text to show to user when they try to withdraw an open course

### Guidance to review
it works?

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally
- [x] Inform data insights team due to database changes

### CODE REVIEW

**Comment by defong** [comment]
[Update withdraw confirm copy for open and closed course](https://trello.com/c/c1O8pSEU/1288-update-withdraw-confirm-copy-for-open-and-closed-course)

File: app/views/publish/courses/withdrawals/edit.html.erb [hunk 1]
```diff
@@ -35,14 +35,14 @@
 
       <p class="govuk-body">Once you’ve withdrawn a course, you cannot publish it again this cycle.</p>
 
-      <h2 class="govuk-heading-m">Confirm withdrawal</h2>
-
-      <%= f.govuk_text_field :confirm_course_code, label: { text: "Enter the course code to confirm" }, width: 5 %>
-
-      <%= f.govuk_submit "Yes I’m sure – withdraw this course", class: "govuk-button govuk-button--warning" %>
+      <% if @course.application_status_open? && FeatureService.enabled?(:open_and_closed_course_flow) %>
+        <p class="govuk-body">If you want to stop new candidates from applying but do not want to withdraw any applications currently in progress, you can <%= govuk_link_to("close the course instead", application_status_publish_provider_recruitment_cycle_course_path) %>.
```
Comment (avinhurry) [inline app/views/publish/courses/withdrawals/edit.html.erb:39]: If you click `close the course instead` then click the back link, you don't go back to the withdraw page.

Shall we have a `goto_withdraw` for the back link?

File: app/controllers/publish/courses/withdrawals_controller.rb [hunk 2]
```diff
@@ -15,7 +15,7 @@ def edit
       def update
         authorize(provider)
 
-        @course_withdrawal_form = CourseWithdrawalForm.new(course, params: withdrawal_params)
+        @course_withdrawal_form = CourseWithdrawalForm.new(course)
```
Comment (avinhurry) [inline app/controllers/publish/courses/withdrawals_controller.rb:18]: Do we want the whole thing behind the feature flag? I don't see any reason to do so.

**Review by avinhurry — COMMENTED** [review]
I think we want to change the second bullet too.

`close applications for it` to `close any applications in progress`

File: app/views/publish/courses/withdrawals/edit.html.erb (hunk 1 above)
Comment (defong) [reply app/views/publish/courses/withdrawals/edit.html.erb]: fixed back link optted with `goto` as the key and `withdraw` as value

File: app/controllers/publish/courses/withdrawals_controller.rb (hunk 2 above)
Comment (defong) [reply app/controllers/publish/courses/withdrawals_controller.rb:18]: Agree, I don't think there is much of a need to put it behind the feature flag

**Pull request merged**